hooks/_common.sh

```diff
@@ -170,6 +170,59 @@ function common::is_hook_run_on_whole_repo {
   fi
 }
 
+#######################################################################
+# Get the number of CPU logical cores available to pre-commit use
+# Arguments:
+#  parallelism_ci_cpu_cores (string) Used in corner cases when CPU cores
+#    can't be calculated
+# Outputs:
+#   Return number CPU logical cores, rounded to below integer
+#######################################################################
+function common::get_cpu_num {
+  local -r parallelism_ci_cpu_cores=$1
+
+  local millicpu
+
+  if [[ -f /sys/fs/cgroup/cpu/cpu.cfs_quota_us ]]; then
+    # Inside K8s pod or DInD in K8s
+    millicpu=$(cat /sys/fs/cgroup/cpu/cpu.cfs_quota_us)
+
+    if [[ $millicpu -eq -1 ]]; then
+      # K8s no limits or in DinD
+      if [[ ! $parallelism_ci_cpu_cores ]]; then
+        common::colorify "yellow" "Unable to calculate available CPU cors.\n" \
+          "You in K8s pod without limits or in DinD without limits propagation.\n" \
+          "To avoid possible harm, parallelism disabled.\n" \
+          "If you'd like reenable it - set corresponding limits, or set next for current hook:\n" \
+          "  args:\n" \
+          "    - --hook-config=--parallelism-ci-cpu-cores=N\n" \
+          "where N is the number of CPU cores you providing to pre-commit."
+
+        return 1
+      fi
+
+      return "$parallelism_ci_cpu_cores"
+    fi
+
+    return $((millicpu / 1000))
+  fi
+
+  if [[ -f /sys/fs/cgroup/cpu.max ]]; then
+    # Inside Linux (Docker?) container
+    millicpu=$(cut -d' ' -f1 /sys/fs/cgroup/cpu.max)
+
+    if [[ "$millicpu" == "max" ]]; then
```

Comment:
Within `[[` you may drop double quotes in this use case:
```suggestion
    if [[ $millicpu == max ]]; then
```